lib/matplotlib/collections.py

```diff
@@ -149,6 +149,7 @@ def __init__(self,
         self._uniform_offsets = None
         if offsets is not None:
             offsets = np.asanyarray(offsets, float)
+            offsets = np.ma.filled(offsets, 0)
```

Comment:
This might happen to fix the scatter example, but I don't understand the logic of it--why should masked values in offsets be set to zero?  I suspect the real problem with scatter and symlog is elsewhere.  The points shouldn't be getting masked in the first place.

Reply:
Hmm, do you remember why you masked them in https://github.com/matplotlib/matplotlib/commit/e959692e42f08c6e157632cf208a6fd19d4228b1 then?

Reply:
Yes.  Originally, scatter handled masked values in any of its array inputs by forming a combined mask and then deleting the masked points from all of the arrays.  The problem was that this led to a mismatch between the original input arrays and the plotted arrays, which caused trouble if the user subsequently tried to modify properties of the resulting collection.  The solution was to let the collection handle the masked points instead of deleting them.
The problem that you are trying to fix here is only revealed, not caused, by this change--it is a problem related to an interaction between symlog and scatter, not a problem with the bad-value handling itself.  It is related to the linear part of the symlog transformation.  The mystery is why it shows up in scatter and not in plot.  The points that are missing are those in the linear range; the default linthresh is 2.0, and the missing points are at 1 and 2.

Reply:
Thanks for the explanation, going down the rabbit hole was driving me a bit nuts! I agree this isn't the right fix here, so will close the PR.